common/txmgr/types/orm.go

```diff
@@ -0,0 +1,95 @@
+package types
+
+import (
+	"context"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+// NEWTX, TX, TXATTEMPT will be converted from generic types to structs at a future date to enforce design and type checks
+type ORM[ADDR any, CHAINID any, HASH any, NEWTX any, R any, TX any, TXATTEMPT any, TXID any, TXMETA any] interface {
```

Comment:
I would prefer the ORM name to be something else, like Persistence or StateManager or something which doesn't mean "Object Relational Mapping".
Mainly because behind this interface, the implementation should be able to use anything, not just a relational DB.

Reply:
Unfortunately I'm not too good at coming up with good names